tests/ssl-opt.sh

```diff
@@ -98,17 +98,25 @@ if [ -z "${GNUTLS_NEXT_SERV:-}" ]; then
 fi
 
 if [ -n "${OPENSSL_NEXT:-}" ]; then
-    O_NEXT_SRV="$OPENSSL_NEXT s_server -www -cert $DATA_FILES_PATH/server5.crt -key $DATA_FILES_PATH/server5.key"
-    O_NEXT_SRV_EARLY_DATA="$OPENSSL_NEXT s_server -early_data -cert $DATA_FILES_PATH/server5.crt -key $DATA_FILES_PATH/server5.key"
-    O_NEXT_SRV_NO_CERT="$OPENSSL_NEXT s_server -www "
-    O_NEXT_CLI="echo 'GET / HTTP/1.0' | $OPENSSL_NEXT s_client -CAfile $DATA_FILES_PATH/test-ca_cat12.crt"
-    O_NEXT_CLI_NO_CERT="echo 'GET / HTTP/1.0' | $OPENSSL_NEXT s_client"
+    O_NEXT_SRV_BASE="$OPENSSL_NEXT s_server -www"
+    O_NEXT_SRV_NO_CERT="$O_NEXT_SRV_BASE -allow_no_dhe_kex"
+    O_NEXT_SRV="$O_NEXT_SRV_NO_CERT -cert $DATA_FILES_PATH/server5.crt -key $DATA_FILES_PATH/server5.key"
+    O_NEXT_SRV_NO_PURE_PSK="$O_NEXT_SRV_BASE -cert $DATA_FILES_PATH/server5.crt -key $DATA_FILES_PATH/server5.key"
+    O_NEXT_SRV_EARLY_DATA="$O_NEXT_SRV -early_data"
+    O_NEXT_CLI_BASE="echo 'GET / HTTP/1.0' | $OPENSSL_NEXT s_client"
+    O_NEXT_CLI_NO_CERT="$O_NEXT_CLI_BASE -allow_no_dhe_kex"
+    O_NEXT_CLI="$O_NEXT_CLI_NO_CERT -CAfile $DATA_FILES_PATH/test-ca_cat12.crt"
+    O_NEXT_CLI_NO_PURE_PSK="$O_NEXT_CLI_BASE -CAfile $DATA_FILES_PATH/test-ca_cat12.crt"
```

Comment:
Just a forewarning, we may need a `O_NEXT_CLI_RENEGOTIATE` if we introduce renegotiation testing. No action required but 
please consider the possible extending the use cases if we are touching this code.